packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,320 @@
+        this.isDragging = true;
+    };
+
+    /**
+     *  Handle mouse moves during a thumb drag operation
+     */
+    private handleMouseMove = (e: MouseEvent): void => {
+        if (this.readOnly || this.disabled || e.defaultPrevented) {
+            return;
+        }
+        // update the value based on current position
+        const eventValue: number =
+            this.orientation === SliderOrientation.horizontal ? e.pageX : e.pageY;
+        this.value = `${this.calculateNewValue(eventValue)}`;
+        this.updateForm();
+    };
+
+    private calculateNewValue = (rawValue: number): number => {
+        // update the value based on current position
+        const newPosition = convertPixelToPercent(
+            rawValue,
+            this.orientation === SliderOrientation.horizontal
+                ? this.trackMinWidth
+                : this.trackMinHeight,
+            this.orientation === SliderOrientation.horizontal
+                ? this.trackWidth
+                : this.trackHeight,
+            this.direction
+        );
+        const newValue: number = (this.max - this.min) * newPosition + this.min;
+        return this.convertToConstrainedValue(newValue);
+    };
+
+    /**
+     * Handle a window mouse up during a drag operation
+     */
+    private handleWindowMouseUp = (event: MouseEvent): void => {
+        this.stopDragging();
+    };
+
+    private stopDragging = (): void => {
+        this.isDragging = false;
+        window.removeEventListener("mouseup", this.handleWindowMouseUp);
+        window.removeEventListener("mousemove", this.handleMouseMove);
+    };
+
+    private clickHandler = (e: MouseEvent) => {
+        if (!this.disabled && !this.readOnly) {
+            this.trackWidth = this.track.clientWidth;
+            if (this.trackWidth === 0) {
+                this.trackWidth = 1;
+            }
+            e.preventDefault();
+            (e.target as HTMLElement).focus();
+            window.addEventListener("mouseup", this.handleWindowMouseUp);
+            window.addEventListener("mousemove", this.handleMouseMove);
+
+            const controlValue: number =
+                this.orientation === SliderOrientation.horizontal ? e.pageX : e.pageY;
+            this.value = `${this.calculateNewValue(controlValue)}`;
+            this.updateForm();
+        }
+    };
+
+    private convertToConstrainedValue = (value: number): number => {
+        const remainderVal: number = value % Number(this.step);
+        const constrainedVal: number =
+            remainderVal >= Number(this.step) / 2
+                ? value - remainderVal + Number(this.step)
+                : value - remainderVal;
+
+        if (constrainedVal < this.min || constrainedVal > this.max) {
+            // TODO here until we figure out how this happens
+            return Number(this.value);
+        } else {
+            return constrainedVal;
+        }
+    };
+
+    private increment = (): void => {
```

Comment:
The `increment` and `decrement` methods seems like nice candidates for public methods that could be invoked in JS.

Reply:
done